src/openrct2-ui/windows/LoadSave.cpp

```diff
@@ -240,17 +240,11 @@ static const char* getFilterPatternByType(const int32_t type, const bool isSave)
 static int32_t window_loadsave_get_dir(const int32_t type, char* path, size_t pathSize)
 {
     const char* last_save = getLastDirectoryByType(type);
-    if (last_save && platform_ensure_directory_exists(last_save))
+    if (platform_directory_exists(last_save))
```

Comment:
last_save can still be nullptr.
```suggestion
    if (last_save != nullptr && platform_directory_exists(last_save))
```

Reply:
When platform_ensure_directory_exists runs, it creates a new empty folder even though the folder was deleted. What i did was that it checks for the existence of a last_save path and that will return a true or false. it will return a false if last_save is a nullptr or doesnt exist anymore, then it shows the user the default path which is
 C:\Documents\OpenRCT2\save

Reply:
That's not what I was talking about. `getLastDirectoryByType()` can return nullptr. This needs to be checked, because `platform_directory_exists()` does not handle nullptr. (That's basic C stuff.)